contrib/kube-prometheus/hack/cluster-monitoring/deploy

```diff
@@ -32,6 +32,12 @@ kctl apply -f manifests/prometheus/prometheus-k8s-service.yaml
 kctl apply -f manifests/alertmanager/alertmanager-config.yaml
 kctl apply -f manifests/alertmanager/alertmanager-service.yaml
 
+# unfortunately statefulsets cannot be changed except for their replica count
+# so we need to make sure that the rule files are created before we create the
+# prometheus resource so it can properly discover the rule files when creating
+# the statefulset
+sleep 5
```

Comment:
Certainly okay for now. But wondering whether we could/should have a sidecar polling configmaps directly and writing them to disk instead of mounting them into pods. Would that get us into trouble with needing more RBACs?